src/utils/synchronization.ts

```diff
@@ -43,29 +43,39 @@ export async function getMetadata(d2: D2, elements: string[]): Promise<Synchroni
 }
 
 export async function postMetadata(instance: Instance, metadata: any): Promise<any> {
-    return await axios.post(instance.url + "/api/metadata", metadata, {
-        auth: {
-            username: instance.username,
-            password: instance.password,
-        },
-        params: {
-            importMode: "COMMIT",
-            identifier: "AUTO",
-            importReportMode: "FULL",
-            importStrategy: "CREATE_AND_UPDATE",
-            mergeMode: "REPLACE",
-            atomicMode: "ALL",
-        },
-    });
+    const params: MetadataImportParams = {
+        importMode: "COMMIT",
+        identifier: "AUTO",
+        importReportMode: "FULL",
+        importStrategy: "CREATE_AND_UPDATE",
+        mergeMode: "REPLACE",
+        atomicMode: "ALL",
+    };
+
+    try {
+        return await axios.post(instance.url + "/api/metadata", metadata, {
+            auth: {
+                username: instance.username,
+                password: instance.password,
+            },
+            params,
+        });
+    } catch (error) {
+        // Catch response error but rethrow unknown issues
+        if (error.response) {
+            return error.response;
+        }
+        throw error;
```

Comment:
Minor. Write explicit `else` to have code in branches. Isn't the comment a bit redundant with the code?